File: mobile/src/app/services/Incident.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Incident} from '../entites/Incident';
import {CustomFilter} from '../entites/CustumFilter';
import {AND, API_URL, INCIDENT_AND_QUERY, INCIDENT_OR_QUERY} from '../utils/constants';
import {SectProvince} from '../entites/SectProvince';
import {SecteurType} from '../entites/SecteurType';
import {ProvinceType} from '../entites/ProvinceType';
import {ProvSecteurType} from '../entites/ProvSecteurType';
import {StatutProvince} from '../entites/StatutProvince';
import {StatutSecteur} from '../entites/StatutSecteur';
import {StatutType} from '../entites/StatutType';
import {BehaviorSubject, Subject} from 'rxjs';
import {StatutSecteurType} from '../entites/StatutSecteurType';

@Injectable({
    providedIn: 'root'
})
export class IncidentService {
    listIncidents: BehaviorSubject<any>;
    constructor(private http: HttpClient) {
        this.listIncidents = new BehaviorSubject<any>(null);
    }
    findAllIncident() {

        return this.http.get(API_URL + '/Incident/find' );

    }
    findIncidentById(id: number) {

        return this.http.get(API_URL + '/Incident/find/' + id );

    }

    AddIncident(incident: Incident) {
        return this.http.post(API_URL + '/Incident/add', incident);
    }

    findByQuery( custmFilter: CustomFilter) {
        // const endpoint = (condition == AND) ? INCIDENT_AND_QUERY : INCIDENT_OR_QUERY;
        return this.http.post(API_URL +  '/Incident/and_query' , custmFilter);
    }
    findByIme(ime: string) {

        return this.http.get(API_URL +  '/Incident/findIme/' + ime );
    }
    findByQueryOr(custmFilter: CustomFilter) {
        return this.http.post(API_URL +  '/Incident/or_query' , custmFilter);
    }

    findByProvince(id: number) {
        return this.http.get(API_URL + '/Incident/find/province/' + id);
    }
    findByBySecteur(id: number) {
        return this.http.get(API_URL + '/Incident/find/secteur/' + id);
    }
    findByByType(id: number) {
        return this.http.get(API_URL + '/Incident/find/type/' + id);
    }
   findByStatut(id: number) {
        return this.http.get(API_URL + '/Incident/find/Statut/'+ id );
    }
    findByProvinceSecteur(sectprovince: SectProvince) {
        return this.http.post(API_URL +  '/Incident/querySectProvince' , sectprovince );
    }
    findBySecteurType(secteurType: SecteurType) {
        return this.http.post(API_URL +  '/Incident/querySectType' ,  secteurType);
    }
    findByProvinceType(provinceType: ProvinceType)
    {
        return this.http.post(API_URL +  '/Incident/queryProvinceType' , provinceType );
    }
    findByProvSecteurTye(provSecteurType: ProvSecteurType)
    {
        return this.http.post(API_URL +  '/Incident/queryProvinceTypeSecteur' , provSecteurType );
    }
    findByStatutProvince(Statutprovince: StatutProvince)
    {
        return this.http.post(API_URL +  '/Incident/querystatutProvince' , Statutprovince );
    }
    findByStatutSecteur(statutSecteur: StatutSecteur)
    {
        return this.http.post(API_URL +  '/Incident/querystatutSecteur' , statutSecteur );
    }
    findByStatutType(statutType : StatutType)
    {
        return this.http.post(API_URL +  '/Incident/querystatutType' , statutType );
    }

    findByStatutSecteurType( statutSecteurType : StatutSecteurType){
        return this.http.post(API_URL +  '/Incident/queryStatutTypeSecteur' , statutSecteurType );
    }





}

File: backend/src/main/java/com/incident/backend/service/IncidentService.java
package com.incident.backend.service;

import java.util.List;

import com.incident.backend.entity.Incident;
import com.incident.backend.entity.Province;
import com.incident.backend.entity.Secteur;
import com.incident.backend.entity.Etat;
import com.incident.backend.entity.Type;
import com.incident.backend.service.helpers.Filter;
import com.incident.backend.service.helpers.IncidentInfos;
import com.incident.backend.service.helpers.ProvinceType;
import com.incident.backend.service.helpers.SecteurProvince;
import com.incident.backend.service.helpers.SecteurType;
import com.incident.backend.service.helpers.StatutProvince;
import com.incident.backend.service.helpers.StatutSecteur;
import com.incident.backend.service.helpers.StatutSecteurType;
import com.incident.backend.service.helpers.StatutType;


public interface IncidentService {
    public void save(Incident incident);
    public void deleteByID(long id );
    public List<Incident> findAll() ;
    public  Incident findByID(long id );
    public List<Incident> findBySecteur(Secteur secteur);
    List<Incident> findByStatut(Etat etat);
    List<Incident> findByProvince(Province province);
    List<Incident> findByType(Type type);
    List<Incident> findByQuery(String field, Object value);
    List<Incident> findByQuerySecteur(Secteur secteur);
    List<Incident> findByQueryCM(Filter customFilter);
    List<Incident> findByQueryBonus(Filter customFilter, String condition);
    List<Incident> findBySecteurProvince(SecteurProvince sectProvince);
    List<Incident> findBySecteurType(SecteurType secteurType);
    List<Incident> findByProvinceType(ProvinceType provinceType);
    List<Incident> finsByProvSectType(IncidentInfos ProvinceSecteurType);
    List<Incident> finsByStatutSectType( StatutSecteurType statutSecteurType);
    List<Incident> findByStatutProv(StatutProvince statutProvince);
     List<Incident> findByStatutSecteur(StatutSecteur StatutSecteur);
     List<Incident> findByStatutType(StatutType statutType);
    List<Incident> findByime(String ime);

   // List findIncidentsPrem();
    List findIncidents();
    List findIncidentsProv();
    List  findIncidentsStatut();
    List  findIncidentsType();
    List<String> getGeometry();
    List findByUserId(long id);

}